CRM/Financial/BAO/FinancialAccount.php

```diff
@@ -246,4 +246,33 @@ public static function getFinancialAccountForFinancialTypeByRelationship($financ
     return Civi::$statics[__CLASS__]['entity_financial_account'][$financialTypeID][$relationTypeId];
   }
 
+  /**
+   * check if financial account is referenced by financial item
+   *
+   * @param $financialAccountId Integer
+   *
+  * @param $financialAccountTypeID Integer
+  *
+   * @return bool
+  *
+  */
+ public static function validateFinancialAccount($financialAccountId, $financialAccountTypeID = NULL) {
+   if (!$financialAccountId) {
```

Comment:
It is an optional parameter. This function is used to validate if Financial Account is used in Financial Item table then its financial account type id cannot be changed. So this function is invoked in 
1. form level to freeze the field if function returns TRUE with $financialAccountTypeID = NULL
2. BAO level when Financial Account is updated i.e when any code snippets tries to update the financial_account_type_id using api or BAO with $financialAccountTypeID = $params['financial_account_type_id']